query/text_parser.js

```diff
@@ -91,4 +102,11 @@ function shouldSetQueryIntoHouseNumber(vs) {
           /^[0-9]+$/.test(vs.var('input:query').toString());
 }
 
+function shouldTreatIslandAsCounty(vs) {
+  return vs.isset('input:island') &&
+    vs.isset('input:region') &&
+    !vs.isset('input:county') &&
+    vs.var('input:region').toString() === 'hawaii';
```

Comment:
Yeah, I don't like this PR for a lot of reasons. One off fixes just generally don't feel right, it doesn't handle other Hawaiian islands/counties in any particularly good way. The only things this PR has going for it is that it fixes an acceptance-test and doesn't seem to break anything else.

You're right that we would have to handle `hi`. I did check and libpostal always lowercases.